internal/schema/schema.go

```diff
@@ -508,26 +582,30 @@ func (s *schemaFetcher) fetchTables(ctx context.Context) ([]Table, error) {
 	for _, _rawTable := range rawTables {
 		rawTable := _rawTable // Capture loop variables for go routine
 		tableFuture, err := concurrent.SubmitFuture(ctx, goroutineRunner, func() (Table, error) {
-			return s.buildTable(ctx, rawTable, tablesToCheckConsMap)
+			return s.buildTable(ctx, rawTable, checkConsByTable)
 		})
 		if err != nil {
 			return nil, fmt.Errorf("starting table future: %w", err)
 		}
 		tableFutures = append(tableFutures, tableFuture)
 	}
+	tables, err := concurrent.GetAll(ctx, tableFutures...)
+	if err != nil {
+		return nil, fmt.Errorf("getting tables: %w", err)
+	}
 
-	return concurrent.GetAll(ctx, tableFutures...)
-}
+	tables = filterSliceByName(
```

Comment:
It is way more flexible to do this filtering in-memory rather than in the query. We can easily support things like regex and complex nested operations in the future (unions of intersections, etc).